### PR TITLE
Concurrency issue

### PR DESCRIPTION
**Pull request by k3rnL**
Hi, 
There is the "rebase" (was easier to fork again..)

[Conversation history here](https://github.com/bytedeco/javacv/pull/1162/)

### CODE REVIEW

File: src/main/java/org/bytedeco/javacv/FFmpegFrameRecorder.java [hunk 1]
```diff
@@ -917,9 +917,7 @@ public boolean recordImage(int width, int height, int depth, int channels, int s
                passing the same picture again */
         } else {
             int step = stride * Math.abs(depth) / 8;
-            BytePointer data = image[0] instanceof ByteBuffer
-                    ? new BytePointer((ByteBuffer)image[0].position(0))
-                    : new BytePointer(new Pointer(image[0].position(0)));
+            BytePointer data = new BytePointer(new Pointer(image[0])).position(0);
```
Comment (saudet) [inline src/main/java/org/bytedeco/javacv/FFmpegFrameRecorder.java]: Leave the branching like it was so it also works with array-based ByteBuffer, at least.

File: src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java [hunk 2]
```diff
@@ -107,25 +107,25 @@ public static int decodeGamma22(int value) {
     public static int encodeGamma22(int value) {
         return gamma22inv[value & 0xFF] & 0xFF;
     }
-    public static void flipCopyWithGamma(ByteBuffer srcBuf, int srcStep,
-            ByteBuffer dstBuf, int dstStep, boolean signed, double gamma, boolean flip, int channels) {
+    public static void flipCopyWithGamma(ByteBuffer srcBuf, int buffer_index, int srcStep,
+                                         ByteBuffer dstBuf, int dst_buffer_index, int dstStep, boolean signed, double gamma, boolean flip, int channels) {
```
Comment (saudet) [inline src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: Let's keep using camel case for the new variables.

File: src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java [hunk 3]
```diff
@@ -409,33 +409,33 @@ public static void flipCopyWithGamma(DoubleBuffer srcBuf, int srcStep,
...
+        applyGamma(frame.image[0], frame.imageDepth, frame.imageStride, gamma);
     }
     public static void applyGamma(Buffer buffer, int depth, int stride, double gamma) {
         if (gamma == 1.0) {
             return;
         }
         switch (depth) {
             case Frame.DEPTH_UBYTE:
-                flipCopyWithGamma(((ByteBuffer)buffer).asReadOnlyBuffer(), stride, (ByteBuffer)buffer, stride, false, gamma, false, 0);
+                flipCopyWithGamma(((ByteBuffer)buffer).asReadOnlyBuffer(), 0, stride, (ByteBuffer)buffer, 0, stride, false, gamma, false, 0);
```
Comment (saudet) [inline src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: The starting indices are not always going to be 0, let's leave them at position() like they were.

Comment (saudet) [reply src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: Or maybe not, since we've been ignoring it, it might be better to keep it that way.

Comment (k3rnL) [reply src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: Actually the method name is not explicit for this use case. Actually I read it as "applyGamma on Buffer with depth, stride and gamma", not "applyGamma on Buffer at this position with depth, stride....."

In any way is not safe to use position as we figured

File: src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java (hunk 2 above)
Comment (saudet) [reply src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: There's still a few more left, please update!

Comment (k3rnL) [reply src/main/java/org/bytedeco/javacv/Java2DFrameConverter.java]: Oh right, didn't noticed them ! 2min..

**Comment by k3rnL** [comment]
Something else wrong ?

**Pull request merged**